XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/Simulation/Hands/RecordedHandJoint.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) XRTK. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using UnityEngine;
+using XRTK.Definitions.Utilities;
+using XRTK.Providers.Controllers.Hands;
+
+namespace XRTK.Providers.Controllers.Simulation.Hands
+{
+    /// <summary>
+    /// Unity's <see cref="JsonUtility"/> "currently" (stated in 2011) does not support top-level arrays.
+    /// But hey, it's on their "future" roadmap. That's why we need a wrapper definition around or items.
+    /// </summary>
+    /// <remarks>The <see cref="JsonUtility"/> also does not support properties, so we gotta use fields. Oh and it also
+    /// does not support auto mapping of JSON naming "items" to C# naming "Items".</remarks>
+    [Serializable]
+    public class RecordedHandJoints
+    {
+        /// <summary>
+        /// Gets the joints recorded in this data set.
+        /// </summary>
+        public RecordedHandJoint[] items;
+    }
+
+    /// <summary>
+    /// A single recorded hand joint's information that may be used to restore the joint pose for simulation.
+    /// </summary>
+    [Serializable]
+    public struct RecordedHandJoint
```

Comment:
I think maybe these types belong in definitions?

Reply:
Agreed, that's the standard we've used through the toolkit to align to the way Unity "prefers" to have definitions defined

Reply:
I don't understand.

Reply:
Create a file for each class or struct @FejZa

Reply:
How'a above comments related to that? Really confusing sometimes. I'll check this but usually I always put everything in its own file. Not a Unity thing but general C# convention.

Reply:
https://github.com/XRTK/XRTK-Core/blob/development/articles/A01-CodingGuidelines.md#public-classes-structs-and-enums-should-all-go-in-their-own-files